### PR TITLE
Component-ize zlux & zss

### PR DESCRIPTION
**Pull request by 1000TurquoisePogs**
See https://github.com/zowe/zlux-app-server/pull/87 for more.
Requires https://github.com/zowe/zlux-app-server/pull/87 and https://github.com/zowe/zlux-build/pull/42

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Necessary documentation (if appropriate) have been added / updated
- [x] DCO signoffs have been added to all commits, including this PR

#### PR type
What type of changes does your PR introduce to Zowe? _Put an `x` in the box that applies to this PR. If you're unsure about any of them, don't hesitate to ask._ 
- [ ] Bugfix 
- [x] Feature 
- [ ] Other... Please describe:

#### Relevant issues
Fixes https://github.com/zowe/zowe-install-packaging/issues/627

#### Does this PR introduce a breaking change?
- [x] Yes
- [ ] No

Many paths & default paths have changed. The behavior of the servers remains the same, but anything relying on specific paths may fail. This is not a drop-in upgrade over a prior release, although manual steps can be taken to do so.

#### Does this PR do something the person installing Zowe should know about?
See https://github.com/zowe/zowe-install-packaging/issues/627

### CODE REVIEW

File: manifest.json.template [hunk 1]
```diff
@@ -12,68 +12,79 @@
   },
   "binaryDependencies": {
     "org.zowe.zlux.zlux-core": {
-      "version": "~1.7.0-STAGING",
+      "version": "~1.8.0-STAGING",
```
Comment (stevenhorsman) [inline manifest.json.template]: It looks like you've not reverted the manifest values to point to the correct staging branches

File: scripts/configure/zowe-configure-iframe-plugins.sh [hunk 2]
```diff
@@ -16,29 +15,24 @@
 
 if [ -z "$NODE_BIN" ]
 then
-  NODE_BIN=${NODE_HOME}/bin/node
+  if [ -n "$NODE_HOME" ]
+  then
+    NODE_BIN=${NODE_HOME}/bin/node
+  else
+    NODE_BIN=node
+  fi
```
Comment (stevenhorsman) [inline scripts/configure/zowe-configure-iframe-plugins.sh]: After configure-node.sh NODE_HOME should always be set, so this shouldn't be needed

File: bin/internal/run-zowe.sh [hunk 3]
```diff
@@ -78,6 +78,17 @@ then
   LAUNCH_COMPONENTS=${LAUNCH_COMPONENTS},files-api,jobs-api,api-mediation,explorer-jes,explorer-mvs,explorer-uss #TODO this is WIP - component ids not finalised at the moment
 fi
 
+#Explorers may be present, but have a prereq on gateway, not desktop
+if [[ $LAUNCH_COMPONENT_GROUPS == *"DESKTOP"* ]]
+then
+  LAUNCH_COMPONENTS=${LAUNCH_COMPONENTS},app-server,zss
```
Comment (stevenhorsman) [inline bin/internal/run-zowe.sh]: Where is components/zss created as it doesn't appear on my, or the Marist install?

**Comment by stevenhorsman** [comment]
When the test are run on Marist the desktop is not accessible and trying to test without zss in the launch component list the following error is in the logs: Encountered parse exception while reading /ZOWE/tmp/.zowe/workspace/app-server/serverConfig/server.json

File: bin/internal/run-zowe.sh (hunk 3 above)
Comment (1000TurquoisePogs) [reply bin/internal/run-zowe.sh]: Turns out I wrote this down but didn't implement zss as a distinct component. It is within app-server. Will remove this and add a note instead.

**Comment by stevenhorsman** [comment]
I've tried testing the latest build and we are still hitting `2020-01-02 15:04:10.893 <ZWED:33620878> IZUSVR WARN (_zsf.json,jsonUtils.js:82) Encountered parse exception while reading /ZOWE/tmp/.zowe/workspace/app-server/serverConfig/server.json`

You can see the rest of the job log https://wash.zowe.org:8443/job/zowe-install-test/job/staging/607/consoleFull

**Comment by 1000TurquoisePogs** [comment]
Hi Steve. This appears to be a bug in the nodejs you are using: 8.16.1
Its similar to https://github.com/ibmruntimes/node/issues/142 but not quite the same...
That issue deals with what to do with untagged files. However, on my system, server.json IS tagged, as iso8850-1. However, it appears that the suggested untagged fix of setting __UNTAGGED_READ_MODE=**V6** somehow also effects **tagged** files, as setting __UNTAGGED_READ_MODE=**DEFAULT** in a test made it work again. However, that breaks **untagged** file behavior for us.

The fix appears to be to tag all of our files, and work was done on that but not merged into 1.8. I propose that for 1.8 we do not use 8.16.1

**Comment by Joe-Winchester** [comment]
This was tested by https://wash.zowe.org:8443/job/zowe-install-test/job/staging/610/.

**Pull request merged**